Allow Symfony 4 components

VolCh opened the pull request:
I am starting to work on support of Symfony 4.0.0-BETA1

VolCh commented:
Added my repos with PRs:
https://github.com/facebook/php-webdriver/pull/489
https://github.com/stecman/symfony-console-completion/pull/79
https://github.com/sebastianbergmann/phpunit/pull/2822 (already merged, repo removed)

Installed symfony components are:
```
$composer info symfony/*
symfony/browser-kit       v4.0.0-BETA1 Symfony BrowserKit Component
symfony/console           v4.0.0-BETA1 Symfony Console Component
symfony/css-selector      v4.0.0-BETA1 Symfony CssSelector Component
symfony/debug             v4.0.0-BETA1 Symfony Debug Component
symfony/dom-crawler       v4.0.0-BETA1 Symfony DomCrawler Component
symfony/event-dispatcher  v4.0.0-BETA1 Symfony EventDispatcher Component
symfony/finder            v4.0.0-BETA1 Symfony Finder Component
symfony/polyfill-mbstring v1.6.0       Symfony polyfill for the Mbstring extension
symfony/process           v4.0.0-BETA1 Symfony Process Component
symfony/yaml              v4.0.0-BETA1 Symfony Yaml Component
```

Feel to free run any tests

DavertMik commented:
Great! However I need to add some guides for this process. 
Dockerfile is not used for testing. What you actually need is to setup `.travis,yml`. We already have similar setup for Symfony 2 and 3 so you can just add one more element to the matrix to run Symfony4 tests. https://github.com/Codeception/Codeception/blob/2.3/.travis.yml#L17

Please change minimal requirements for composer.json but change them dynamically in Travis
github.com/Codeception/Codeception/blob/2.3/.travis.yml#L46

VolCh commented:
Current status:

Waiting merge & tag PR https://github.com/stecman/symfony-console-completion/pull/79
Waiting merge (at least) PR https://github.com/symfony/symfony/pull/24975 causes TypeError in tests
Investigating other tests failures & symfony/symfony:4.0.0 composer conflict

DavertMik commented:
Ok, we have only one blocker: 

stecman/symfony-console-completion#79

If we don't see it merged in few next days we can remove it from required dependencies and allow it to be installed optionally. This feature is not that important... But we need to keep it going

mariusbalcytis commented:
Few days have passed - should we still wait? Unfortunately this cascades to other libraries and projects - they cannot be updated for sf4, as their tests cannot pass (if they are written in codeception). Thus other libraries and projects cannot be updated and so on.

DavertMik commented:
> Few days have passed - should we still wait? Unfortunately this cascades to other libraries and projects - they cannot be updated for sf4, as their tests cannot pass (if they are written in codeception). Thus other libraries and projects cannot be updated and so on.

Recursion! Yes, I think it can be merged. 
But before the release I'd detach symfony autocompletion package as it is not updateding...

Naktibalda commented on tests/cli/DataProviderFailuresAndExceptionsCest.php:
```diff
@@ -43,7 +43,6 @@ public function runTestWithDataProvidersFailureStdout(\CliGuy $I)
     public function runTestWithDataProvidersFailureStderr(\CliGuy $I)
     {
         $I->executeCommand('run -n unit DataProvidersFailureCest 2>&1',false);
-        $I->seeInShellOutput('[Codeception\Exception\TestParseException]');
```
Why did you remove some assertions from this test?

DavertMik replied:
The format of output in Symfony Console has changed

VolCh commented:
Added fallback for absent stecman/symfony-console-completion

Rebased and splitted to atomic commits.

I think it ready to work in Symfony 4 environment. Needs some CI changes after Symfony 4 release.

Full support of Symfony 4 framework (its DI container first of all) I am planing check & implement at weekend in another PR.

DavertMik commented:
Thank you!
